modules/azure/monitoring_metric_alert/variables.tf

```diff
@@ -19,9 +19,16 @@ variable "monitoring_rules" {
     aggregation             = string
     operator                = string
     threshold               = number
+    frequency               = optional(string)
+    window_size             = optional(string)
     auto_mitigation_enabled = optional(bool)
     severity                = optional(number)
     enabled                 = optional(bool)
+    dimension = optional(object({
```

Comment:
Tested this if `dimension` is not set or set to null, throws error

```
╷
│ Error: Attempt to get attribute from null value
│ 
│   on main.tf line 15, in data "null_data_source" "test_for_each_set":
│   15:     name = each.value.dimension.values
│     ├────────────────
│     │ each.value.dimension is null
│ 
│ This value is null, so it does not have any attributes.
╵
```

Reply:
My bad, fixed made the block dynamic, should be ok now

Reply:
It works but we could also make it an optional list, because the specs allow multiple dimension blocks https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_metric_alert#dimension-1

```terraform
dynamic "dimension" {
  for_each = each.value.dimension
  content {
    name     = dimension.value.name
    operator = dimension.value.operator
    values   = dimension.value.values
  }
}
```

```terraform
dimension = optional(list(object({
  name     = string,
  operator = string,
  values   = list(string)
})), [])
```

Just an idea.

Reply:
True, allowed a list of dimensions